lib/ash/resource/transformers/validate_validation_attributes.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Ash.Resource.Transformers.ValidateValidationAttributes do
+  @moduledoc """
```

Comment:
I think this will be too generic, as there are some cases where validation modules may use the key `attributes` but may actually support additional things (i.e arguments). This would also check any non-built in validations and we can't make any assertions about what the `attributes` key means for them.

In the long run, I'd like to solve this via giving validations away to validate their options against a resource. Currently validations have an `init/1` callback, which is meant to validate options, but I'm thinking that we should allow them to have an `init/2` callback that takes the options and the resource and validates them. However, we can't do this right now because that would cause compilations issues, which we need to figure out a way around.

In the short term, I think this validation needs to be run against only specific, known validations, and those validations should be double-checked to ensure that they actually only accept attributes.

Additionally, we should check the validations on the action, which are in the `.changes` key of each action (would need to filter out the ones that are validations.